Remove handling account ID in app, prevent changing settings when signed in

roop opened the pull request:
This PR

  - Addresses the Apple clients part of #2791 
  - Fixes #2668

The Settings UI change of separating Logs and Advanced settings is not part of this PR.

roop committed:
TunnelStore: Fix "stale configuration" error when saving the tunnel configuration

By reloading the configuration before modifying it

roop committed:
SettingsView: Show confirmation alert when changing settings while signed in

roop committed:
Remove handling Account ID in the app

Account ID shall be handled in the web portal instead

roop committed:
SettingsView: Sign out the user when saving advanced settings

roop committed:
Cleanup TunnelAuthStatus

- Remove .accountNotSetup state
- Remove authBaseURL data from signed out state because
  that's part of advanced settings anyway

roop committed:
Add some logging

roop committed:
AuthStore: Bring tunnelAuthStatus in line, in case it's out of touch with the login status

roop committed:
AuthStore: Fix handling of TunnelShutdownEvent

Since tunnelStore.start() now returns only after one of these happens:
  - tunnel.start() throws
  - tunnel status becomes .connected
  - tunnel status becomes .disconnected

the TunnelShutdownEvent can be handled when tunnelStore.start() returns.

roop committed:
TunnelStore: If accountId was stored, remove it

roop committed:
Fix iOS build

jamilbk commented on swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift:
```diff
@@ -176,95 +164,50 @@ public struct SettingsView: View {
+          }
+          Button("Cancel", role: .cancel) {
+            // Nothing to do
+          }
```
Does order matter here?

```suggestion
          Button("Cancel", role: .cancel) {
            // Nothing to do
          }
          Button("Continue") {
            confirmationAlertContinueAction.performAction(on: self)
          }
```

roop replied:
It doesn't. SwiftUI will take care of placing the cancel-role-button to the left. Nevertheless, I agree it's a good idea to change the order in the code anyway.

roop replied:
Fixed in commit c9d6145.

jamilbk commented on swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift:
```diff
@@ -176,95 +164,50 @@ public struct SettingsView: View {
+          }
+          Button("Cancel", role: .cancel) {
+            // Nothing to do
+          }
```
```suggestion
          Button("Cancel", role: .cancel) {
            // Nothing to do
          }
          Button("Continue") {
            confirmationAlertContinueAction.performAction(on: self)
          }
```

roop replied:
Fixed in commit c9d6145.

jamilbk commented on swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/TunnelStore.swift:
```diff
@@ -12,11 +12,14 @@ import OSLog
 public struct TunnelProviderKeys {
   static let keyAuthBaseURLString = "authBaseURLString"
-  static let keyAccountId = "accountId"
+  static let keyAccountId = "accountId"  // Unused
```
Can this just be removed if we're not using it?
```suggestion
```

roop replied:
We use it to remove the data from the tunnel configuration for tunnel configurations created by older versions of the app. I will clarify the comment.

roop replied:
Fixed in commit e2e79d7.

jamilbk reviewed (changes requested):
Just a question on the button order and minor cleanup otherwise LGTM

roop committed:
SettingsView: Change the order of buttons in the code

SwiftUI places the cancel button to the left irrespective of this order,
but it's better that our code is consistent with the UI.

roop committed:
Clarify comment about keyAccountId
